File: 617_MergeTwoBinaryTrees/Jason/617.cpp
#include <iostream>

struct TreeNode {
	int val;
 	TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
	if (t1 == NULL)
		return t2;
	
	if (t2 == NULL)
		return t1;

	TreeNode *tmp = new TreeNode(
		t1->val + t2->val
	);
	
	tmp->left = mergeTrees(t1->left, t2->left);
	tmp->right = mergeTrees(t1->right, t2->right);
	
	return tmp;
}

File: 617_MergeTwoBinaryTrees/Jason/617.go
package main

func main() {

}

func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	// Version 1
	if t1 == nil {
		return t2
	}

	if t2 == nil {
		return t1
	}

	return &TreeNode{
		Val:   t1.Val + t2.Val,
		Left:  mergeTrees(t1.Left, t2.Left),
		Right: mergeTrees(t1.Right, t2.Right),
	}
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}
